Backpack/Price/Classes/BPKPrice.swift

```diff
@@ -0,0 +1,203 @@
+    public enum Size {
+        case large, small
+    }
+    
+    public enum Alignment {
+        case left, right
+    }
+    
+    public var price: String {
+        didSet {
+            priceLabel.text = price
+        }
+    }
+    
+    public var leadingText: String? {
+        didSet {
+            leadingTextLabel.text = leadingText
+        }
+    }
+    
+    public var lineThroughText: String? {
+        didSet {
+            applyLineThroughStyling()
+        }
+    }
+    
+    public var alignment: Alignment
+    public let size: Size
+    
+    public var trailingText: String? {
+        didSet {
+            trailingTextLabel.text = trailingText
+        }
+    }
+    
+    private lazy var strikeThroughTextAttributes: [NSAttributedString.Key: Any] = {
+        [
+            .foregroundColor: BPKColor.textSecondaryColor,
+            .font: BPKFont.makeFont(fontStyle: footnoteOrCaptionFontStyle()),
+            .strikethroughStyle: NSUnderlineStyle.single.rawValue,
+            .strikethroughColor: BPKColor.textSecondaryColor
+        ]
+    }()
+    
+    private lazy var priceLabel: BPKLabel = {
+        let label = BPKLabel(fontStyle: size == .large ? .textHeading2 : .textHeading4)
+        return label
+    }()
+    
+    private lazy var trailingTextLabel: BPKLabel = {
+        let label = BPKLabel(fontStyle: footnoteOrCaptionFontStyle())
+        label.textColor = BPKColor.textSecondaryColor
+        return label
+    }()
+    
+    private lazy var lineThroughTextLabel: BPKLabel = {
+        let label = BPKLabel(fontStyle: footnoteOrCaptionFontStyle())
+        label.textColor = BPKColor.textSecondaryColor
+        return label
+    }()
+    
+    private lazy var separatorLabel: BPKLabel = {
+        let label = BPKLabel(fontStyle: footnoteOrCaptionFontStyle())
+        label.textColor = BPKColor.textSecondaryColor
+        return label
+    }()
+    
+    private lazy var leadingTextLabel: BPKLabel = {
+        let label = BPKLabel(fontStyle: footnoteOrCaptionFontStyle())
+        label.textColor = BPKColor.textSecondaryColor
+        return label
+    }()
+    
+    private lazy var topTextStackView: UIStackView = {
+        let stackView = UIStackView(arrangedSubviews: [lineThroughTextLabel, separatorLabel, leadingTextLabel])
+        stackView.axis = .horizontal
+        stackView.spacing = BPKSpacingSm
+        return stackView
+    }()
+    
+    private lazy var priceStackView: UIStackView = {
+        let stackView = UIStackView(arrangedSubviews: [priceLabel, trailingTextLabel])
+        stackView.axis = .horizontal
+        stackView.spacing = BPKSpacingSm
+        stackView.alignment = .firstBaseline
+        return stackView
+    }()
+    
+    private lazy var containerStackView: UIStackView = {
+        let stackView = UIStackView(arrangedSubviews: [topTextStackView, priceStackView])
+        stackView.axis = .vertical
+        return stackView
+    }()
+    
+    public init(
+        price: String,
```

Comment:
What's expected to be set as price when a viewcontroller's being initialised and the data hasn't come from the backend yet? 
Forcing this in the initialiser will force clients to set dummy values until the data comes.
You can compare this component with the regular `UILabel`'s init